### ISSUE TITLE
OME not recognizing self signed certificate

### ISSUE THREAD

**Issue by dimiden** [AirenSoft/OvenMediaEngine#1318]
### Discussed in https://github.com/AirenSoft/OvenMediaEngine/discussions/1317

<div type='discussions-op-text'>

<sup>Originally posted by **Shutoparu** July 18, 2023</sup>
I am using a self signed certificate on OME, but this keeps showing in the log (docker environment):
```
[2023-07-18 03:55:30.568] E [SPAPISvr-t8082:31] OpenSSL | tls.cpp:208  | An error occurred while accept SSL connection: [OpenSSL] error:0A000418:SSL routines::tlsv1 alert unknown ca (167773208)
```
I am using the following Server.xml. I'm not sure how can I get the chained certificate from a self signed certificate, so for now I simply put the same certificate in there. (pretty sure this is where it goes wrong.)
```
<?xml version="1.0" encoding="UTF-8"?>

<Server version="8">
    <Name>OvenMediaEngine</Name>
    <Type>origin</Type>
    <IP>*</IP>
    <!-- <IP>::</IP> -->
    <StunServer>stun.l.google.com:19302</StunServer>
    <Bind>
        <Providers>
            <RTMP>
                <Port>1935</Port>
            </RTMP>
            <SRT>
                <Port>9999</Port>
                <WorkerCount>1</WorkerCount>
            </SRT>
            <WebRTC>
                <Signalling>
                    <Port>3333</Port>
                    <TLSPort>3334</TLSPort>
                </Signalling>
                <IceCandidates>
                    <TcpRelay>*:3478</TcpRelay>
                    <!-- <TcpRelay>[::]:3478</TcpRelay> -->
                    <IceCandidate>*:10000-10005/udp</IceCandidate>
                    <!-- <IceCandidate>[::]:10000-10005/udp</IceCandidate> -->
                </IceCandidates>
            </WebRTC>
        </Providers>
        <Publishers>
            <WebRTC>
                <Signalling>
                    <Port>3333</Port>
                    <TLSPort>3334</TLSPort>
                </Signalling>
                <IceCandidates>
                    <TcpRelay>*:3478</TcpRelay>
                    <!-- <TcpRelay>[::]:3478</TcpRelay> -->
                    <IceCandidate>*:10000-10005/udp</IceCandidate>
                    <!-- <IceCandidate>[::]:10000-10005/udp</IceCandidate> -->
                </IceCandidates>
            </WebRTC>
            <LLHLS>
                <Port>3333</Port>
                <TLSPort>3334</TLSPort>
                <WorkerCount>1</WorkerCount>
            </LLHLS>
        </Publishers>
        <Managers>
            <API>
                <Port>8081</Port>
                <TLSPort>8082</TLSPort>
            </API>
        </Managers>
    </Bind>
    <Managers>
        <Host>
            <Names>
                <Name>*</Name>
            </Names>
            <TLS>
                <CertPath>/etc/cert/server.crt</CertPath>
                <ChainCertPath>/etc/cert/server.crt</ChainCertPath>
                <KeyPath>/etc/cert/server.key</KeyPath>
            </TLS>
        </Host>
        <API>
            <AccessToken>admin:admin</AccessToken>
        </API>
    </Managers>

    <VirtualHosts>
        <!-- You can use wildcard like this to include multiple XMLs -->
        <VirtualHost include="VHost*.xml" />
        <VirtualHost>
            <Name>default</Name>
            <Host>
                <Names>
                    <Name>*</Name>
                </Names>
                <TLS>
                    <CertPath>/etc/cert/server.crt</CertPath>
                    <ChainCertPath>/etc/cert/server.crt</ChainCertPath>
                    <KeyPath>/etc/cert/server.key</KeyPath>
                </TLS>
            </Host>
            <!-- Settings for applications -->
            <Applications>
                <Application>
                    <Name>app</Name>
                    <!-- Application type (live/vod) -->
                    <Type>live</Type>
                    <OutputProfiles>
                        <OutputProfile>
                            <Name>bypass_stream</Name>
                            <OutputStreamName>${OriginStreamName}</OutputStreamName>
                            <Encodes>
                                <Video>
                                    <Bypass>true</Bypass>
                                </Video>
                                <Audio>
                                    <Bypass>true</Bypass>
                                </Audio>
                                <Audio>
                                    <Codec>opus</Codec>
                                    <Bitrate>128000</Bitrate>
                                    <Samplerate>48000</Samplerate>
                                    <Channel>2</Channel>
                                </Audio>
                            </Encodes>
                        </OutputProfile>
                    </OutputProfiles>
                    <Providers>
                        <WebRTC />
                        <RTMP />
                        <SRT />
                    </Providers>
                    <Publishers>
                        <AppWorkerCount>1</AppWorkerCount>
						<StreamWorkerCount>8</StreamWorkerCount>
                        <WebRTC>
                            <Timeout>30000</Timeout>
                            <Rtx>false</Rtx>
                            <Ulpfec>false</Ulpfec>
                        </WebRTC>
                        <LLHLS>
                            <ChunkDuration>0.5</ChunkDuration>
                            <SegmentDuration>6</SegmentDuration>
                            <SegmentCount>10</SegmentCount>
                            <CrossDomains>
                                <Url>*</Url>
                            </CrossDomains>
                        </LLHLS>
                    </Publishers>
                </Application>
            </Applications>
        </VirtualHost>
    </VirtualHosts>
</Server>
```

For the certificate, this is the command I ran (on MacOS):
```
openssl req -x509 -new -nodes -sha256 -utf8 -days 3650 -newkey rsa:2048 -keyout server.key -out server.crt -config ssl.conf
```
ssl.conf:
```
[req]
prompt = no
default_md = sha256
default_bits = 2048
distinguished_name = dn
x509_extensions = v3_req

[dn]
C = TW
ST = Taiwan
L = Taipei
O = Shutoparu
OU = Shuto test
emailAddress = my@email.com
CN = localhost

[v3_req]
subjectAltName = @alt_names

[alt_names]
DNS.1 = *.localhost
DNS.2 = localhost
IP.1 = 192.XXX.XXX.XXX
```

I'm new to all these stuff so any help is appreciated. Thanks in advance.</div>

**Comment by dimiden** [comment]
@Shutoparu
This issue occurred due to an incorrect certificate set in the ```<ChainCertPath>```.
As mentioned in #1317, there was an issue with errors occurring when ```<ChainCertPath>``` is not set, and it seems like you set ```server.crt``` due to this issue.
I have just fixed this and committed f3cc17ecc097e1c798451eaf07494ab8197a7643.

**Comment by getroot** [comment]
The fix for this bug is part of the 0.15.14 release.
